backend/inspirehep/records/api/mixins.py

```diff
@@ -228,6 +346,46 @@ def update(self, data, disable_relations_update=False, *args, **kwargs):
         else:
             self.update_refs_in_citation_table()
 
+    def get_all_connected_papers_of_modified_authors(self):
+        prev_version = self._previous_version
+        current_authors = set(self.get_authors_bais())
+        old_authors = set(prev_version.get_authors_bais())
+        diff = list(current_authors.symmetric_difference(old_authors))
+        differed_papers_uuids = []
+        if diff:
+            differed_papers_uuids = [
+                result.record_id
+                for result in RecordsAuthors.query.filter(
+                    RecordsAuthors.record_id != self.id
+                )
+                .filter(
+                    and_(RecordsAuthors.author_id.in_(diff)),
+                    RecordsAuthors.id_type == "INSPIRE BAI",
+                )
+                .all()
+            ]
+        return differed_papers_uuids
```

Comment:
Not sure if I'm stupid or this is crazy, but this looks like you're going to reindex all ATLAS papers as soon as a single ATLAS paper is touched. A self-citation is a citation, so it can only affect citing or cited papers. Cited papers get reindexed already in any case to update the citation count, so that leaves citing papers. Can't you use a query on the citation table you've updated previously to find those?

Reply:
Hmm. This is how I understand this requirement:
```
 Change get modified references to return also for changed authors or collaborations.
```
But it might be too wide, you are right.

Reply:
Ah, ok I get it.

Reply:
:+1: